File: sfml demos/C++/exercise 04/SFML/Player.cpp
#include "Player.h"

Player::Player() : ActionTarget(_playerInputs), _shape(Vector2f(32, 32)), _isMoving(false)
{
	_shape.setFillColor(Color::Blue);
	_shape.setOrigin(16, 16);

	bind(PlayerInputs::Up, [this](const Event&)
	{
		_isMoving = true;
	});

	bind(PlayerInputs::Left, [this](const Event&)
	{
		_isMoving = true;
	});

	bind(PlayerInputs::Right, [this](const Event&)
	{
		_isMoving = true;
	});

	bind(PlayerInputs::Down, [this](const Event&)
	{
		_isMoving = true;
	});
}

void Player::processEvents()
{
	_isMoving = false;
	ActionTarget::processEvent();
}

void Player::setDefaultInputs()
{
	_playerInputs.map(PlayerInputs::Up, Action(Keyboard::Up));
	_playerInputs.map(PlayerInputs::Right, Action(Keyboard::Right));
	_playerInputs.map(PlayerInputs::Left, Action(Keyboard::Left));
	_playerInputs.map(PlayerInputs::Down, Action(Keyboard::Down));
}

File: sfml demos/SFMLFramework - master/SFMLFramework/src/Player.cs
using SFML.System;
using SFML.Window;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class Player : Actor
{
    public Mover mover;
    public Renderer renderer;
    public PlayerKeyboardController keyboardController;
    public ICollision collider;
    public CollisionRender collisionRender;
    public Rigidbody rigidbody;

    private string spriteSheedPath = "dragon.png";

    public Player()
    {
        this.name = "Player 1";

        this.mover = AddComponent<Mover>();
        this.mover.Speed = new Vector2f(200, 200);

        this.renderer = AddComponent<Renderer>();
        this.renderer.iMove = this.mover;
        this.renderer.LoadSpriteSheet(spriteSheedPath);

        this.keyboardController = AddComponent<PlayerKeyboardController>();
        this.keyboardController.keyPressedActions.Add(
                Keyboard.Key.A, new Action(() =>
                    this.mover.SetDirectionMove(Mover.EDirection.Left, true)));
        this.keyboardController.keyPressedActions.Add(
                Keyboard.Key.D, new Action(() =>
                   this.mover.SetDirectionMove(Mover.EDirection.Right, true)));
        /*
        this.keyboardController.keyPressedActions.Add(
                Keyboard.Key.W, new Action(() =>
                   this.mover.SetDirectionMove(Mover.EDirection.Up, true)));
        this.keyboardController.keyPressedActions.Add(
                Keyboard.Key.S, new Action(() =>
                   this.mover.SetDirectionMove(Mover.EDirection.Down, true)));
        */

        this.keyboardController.keyReleasedActions.Add(
                Keyboard.Key.A, new Action(() =>
                   this.mover.SetDirectionMove(Mover.EDirection.Left, false)));
        this.keyboardController.keyReleasedActions.Add(
                Keyboard.Key.D, new Action(() =>
                   this.mover.SetDirectionMove(Mover.EDirection.Right, false)));
        /*
        this.keyboardController.keyReleasedActions.Add(
                Keyboard.Key.W, new Action(() =>
                    this.mover.SetDirectionMove(Mover.EDirection.Up, false)));
        this.keyboardController.keyReleasedActions.Add(
                Keyboard.Key.S, new Action(() =>
                    this.mover.SetDirectionMove(Mover.EDirection.Down, false)));
        */

        this.collider = AddComponent<RectCollider>();
        this.collider.SetSprite(this.renderer.SpriteSheet.Sprite);
        this.collider.IMove = this.mover;

        this.collisionRender = AddComponent<CollisionRender>();
        this.collisionRender.shape = this.collider.GetShape();

        this.rigidbody = AddComponent<Rigidbody>();
        this.rigidbody.mover = mover;
        this.rigidbody.mass = 2;
    }    
}
